x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java

```diff
@@ -321,7 +321,19 @@ public void testFailsIfRegisterHoldsSpuriousValue() {
             @Override
             public long onCompareAndExchange(AtomicLong register, long expected, long updated) {
                 if (randomBoolean() && sawSpuriousValue.compareAndSet(false, true)) {
-                    return randomFrom(expectedMax, randomLongBetween(expectedMax, Long.MAX_VALUE), randomLongBetween(Long.MIN_VALUE, -1));
+                    if (register.get() == expectedMax) {
+                        return randomFrom(
+                            randomLongBetween(0L, expectedMax - 1),
+                            randomLongBetween(expectedMax + 1, Long.MAX_VALUE),
+                            randomLongBetween(Long.MIN_VALUE, -1)
```

Comment:
NIT: I think `randomValueOtherThan(expectedMax, ESTestCase::randomLong)` would do the same.

Reply:
Kinda, except here I am deliberately choosing the "likely" wrong values with much higher probability.